apps/web/app/(app)/environments/[environmentId]/actions.ts

```diff
@@ -208,13 +208,16 @@ export async function copyToOtherEnvironmentAction(
   return newSurvey;
 }
 
-export const deleteSurveyAction = async (surveyId: string) => {
+export const deleteSurveyAction = async (surveyId: string, environmentId: string) => {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
 
   const isAuthorized = await canUserAccessSurvey(session.user.id, surveyId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasDeleteAccess } = await verifyUserRoleAccess(environmentId, session.user.id);
```

Comment:
we have the same action problem as mentioned below. Please derive environmentId instead of param.